engine/src/main/java/org/hibernate/search/bridge/util/impl/NumericFieldUtils.java

```diff
@@ -99,11 +100,10 @@ public static boolean requiresNumericRangeQuery(Object value) {
 		if ( value == null ) {
 			return false;
 		}
-		final Class<?> numericClass = value.getClass();
-		return numericClass.isAssignableFrom( Double.class ) ||
-				numericClass.isAssignableFrom( Long.class ) ||
-				numericClass.isAssignableFrom( Integer.class ) ||
-				numericClass.isAssignableFrom( Float.class ) ||
-				numericClass.isAssignableFrom( Calendar.class );
+		return value instanceof Double ||
+				value instanceof Long ||
+				value instanceof Integer ||
+				value instanceof Float ||
+				value instanceof Calendar;
```

Comment:
Hmmm, I'm surprised this is not consistent with the code above. Shouldn't we also have Byte, Date and Short here?